### PR TITLE
Provide language selection as URL parameter

### LINKED ISSUE #154: provide language selection as URL parameter
We should have the possibility to provide the wanted language for the FAQ via URL parameter. If that were available, we could invoke the FAQ from inside cgeo using the language of the cgeo app. That way a German user would immediately land on the German version of the FAQ.

### PR DESCRIPTION
**Pull request by shroudedcode**
This should close #154.

### CODE REVIEW

File: faq.html [hunk 1]
```diff
@@ -97,8 +97,15 @@
...
 <script type="text/javascript">
-    $('#faq-language-selection').show();
-
-    var userLang = (navigator.language || navigator.userLanguage).split('-')[0];
-    switchLanguage(userLang);
+    $('#faq-language-selection').show()
+    const parameters = location.search
+        .substr(1)
+        .split("&")
+        .reduce((obj, str) => {
```
Comment (SammysHP) [inline faq.html]: While I like the arrow function syntax, it is not supported in enough browsers AFAIK.

File: faq.html [hunk 2]
```diff
@@ -13,7 +13,7 @@
     const languages = ["{{ languages | join: '", "' }}"]
 
     function switchLanguage(lang) {
-        if ($.inArray(userLang, languages) > -1) {
+        if (languages.includes(lang)) {
```
Comment (SammysHP) [inline faq.html]: This is easier to understand, but I'm not sure about the browser compatibility. Most modern browsers should support it, though.

File: faq.html (hunk 1 above)
Comment (shroudedcode) [reply faq.html]: [image](http://caniuse.com/#search=arrow%20function)

It seems like it's supported pretty well by most modern browsers.

Comment (SammysHP) [reply faq.html]: Except of the mobile area, especially older stock browsers and Opera Mini.

I just tested our current version with Opera Mini and it doesn't work either. OK, let's not support Opera Mini. ;)

**Comment by SammysHP** [comment]
I had a look at some of the jekyll multi-language modules. They generate paths like `cgeo.org/de/faq` and `cgeo.org/en/faq` and maybe a redirect/default at `cgeo.org/faq`. This would make this a lot easier because it works also without javascript.

**Comment by shroudedcode** [comment]
@SammysHP Can we use these modules? Or do you think about manually recreating the URLs?

**Comment by SammysHP** [comment]
These modules would create everything automatically. We would only need to create files with the translated strings and use string references in the layouts. See #155.

**Comment by shroudedcode** [comment]
@SammysHP We would have to wait for GitHub Pages to allow these modules, right?

**Comment by SammysHP** [comment]
Yes, but there is already an issue for that and I hope that they will implement it in the near future.

**Comment by Lineflyer** [comment]
@SammysHP 
What is the current status of this PR?
I see two issues refering to it: #153 and #156 

I do not see an advantage in #156 as (like you said) the language is selected automatically, while #153 should be solved.
So from my point of view a fix for #153 should be enough and we can close #156 without changes.

**Pull request closed without merging**